source/testcommon/tests/conftest.py

```diff
@@ -0,0 +1,42 @@
+from typing import Generator
+import pytest
+
+from pyspark.sql import SparkSession
+from delta import configure_spark_with_delta_pip
+
+
+@pytest.fixture(scope="module", autouse=True)
+def clear_cache(spark: SparkSession):
+    yield
+    # Clear the cache after each test module to avoid memory issues
+    spark.catalog.clearCache()
+
+
+@pytest.fixture(scope="session")
+def spark() -> Generator[SparkSession, None, None]:
+    warehouse_location = "__spark-warehouse__"
+
+    session = configure_spark_with_delta_pip(
+        SparkSession.builder.config("spark.sql.warehouse.dir", warehouse_location)
+        .config("spark.sql.streaming.schemaInference", True)
+        .config("spark.ui.showConsoleProgress", "false")
+        .config("spark.ui.enabled", "false")
+        .config("spark.ui.dagGraph.retainedRootRDDs", "1")
+        .config("spark.ui.retainedJobs", "1")
+        .config("spark.ui.retainedStages", "1")
+        .config("spark.ui.retainedTasks", "1")
+        .config("spark.sql.ui.retainedExecutions", "1")
+        .config("spark.worker.ui.retainedExecutors", "1")
+        .config("spark.worker.ui.retainedDrivers", "1")
+        .config("spark.default.parallelism", 1)
+        .config("spark.driver.memory", "2g")
+        .config("spark.executor.memory", "2g")
+        .config("spark.rdd.compress", False)
+        .config("spark.shuffle.compress", False)
+        .config("spark.shuffle.spill.compress", False)
+        .config("spark.sql.shuffle.partitions", 1)
+        .enableHiveSupport()
```

Comment:
Is hive needed?

Reply:
I just copied the one from wholesale and didn't look back. I've made this barebone now :)